client/src/pages/Donations.jsx

```diff
@@ -1,36 +1,180 @@
+import React, { useState, useEffect, useContext, useCallback } from "react";
 import axios from "axios";
-import { Organization } from "../components/Organization";
-import { Container, Row, Table} from "react-bootstrap";
+import { Container, Table, Form, Button } from "react-bootstrap";
 import { UserContext } from "../utils/auth";
 
 export const Donations = () => {
-    const [contributions, setContributions] = useState({"Feeding America":"$20", "Ocean Conservacy":"$20","Education Through Music":"$20","Direct Relief":"$20"});
+  const { user } = useContext(UserContext);
+  const [donationData, setDonationData] = useState({ name: "", amount: 0 });
+  const [errors, setErrors] = useState({});
+  const [contributions, setContributions] = useState([]);
 
-    const addContributions = () => {
-        
-    }
+  const refreshDonations = useCallback(async () => {
+    const res = await axios.get("/api/donations", {
+      params: {
+        userId: user.uid,
+      },
+    });
+    setContributions(res.data);
+  }, [user.uid]);
+
+  //get all donations that the user currently has
+  useEffect(() => {
+    refreshDonations();
+  }, [user, refreshDonations]);
+
+  // finds form errors
+  const findFormErrors = () => {
+    const { name, amount } = donationData;
+    const newErrors = {};
+    // name errors
+    if (name === "") newErrors.name = "cannot be blank!";
+    // donation errors
+    if (amount === 0 || amount === NaN || amount === "")
+      newErrors.amount = "cannot be blank!";
 
-    const generateContributions = () => {
-        let temp = [];
-        for (const [key, value] of Object.entries(contributions)) {
-            temp.push(<tr><td>{key}</td><td>{value}</td></tr>);
-            console.log(key, value);
-        }
-        return temp;
+    return newErrors;
+  };
+
+  const handleChange = (e) => {
+    setDonationData({
+      ...donationData,
+      [e.target.name]: e.target.value,
+    });
+    // Check and see if errors exist, and remove them from the error object:
+    if (!!errors[e.target.name])
+      setErrors({
+        ...errors,
+        [e.target.name]: null,
+      });
+  };
+
+  //should make an axios call to add a donation to the user
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const newErrors = findFormErrors();
+    if (Object.keys(newErrors).length > 0) {
+      // We got errors!
+      setErrors(newErrors);
+    } else {
+      // No errors! Put any logic here for the form submission!
+      await axios.post("/api/donations", null, {
+        params: {
+          orgName: donationData.name,
+          donationAmount: donationData.amount,
+          userId: user.uid,
+        },
+      });
+      refreshDonations();
     }
+  };
+
+  //pulls id from svg tag and then refreshes donations
+  const handleDelete = async (e) => {
+    await axios.delete("/api/donations/" + e.target.id, {
+      params: {
+        donationId: e.target.id,
+      },
+    });
+    refreshDonations();
+  };
```

Comment:
```javascript
const handleDelete = async (id) => {
  await axios.delete("/api/donations/" + id, {
    params: {
      donationId: id,
    },
  });
  refreshDonations();
};

/* ....*/

<svg
  xmlns="http://www.w3.org/2000/svg"
  width="16"
  height="16"
  fill="currentColor"
  className="bi bi-x-circle"
  viewBox="0 0 16 16"  
  onClick={() => {
    handleDelete(contribution.id);
  }}
  style={{ cursor: "pointer" }}
>
```

Since we are using this `id` value to communicate with the backend, we probably don't want to pick it up from event `target` since the `id` grabbed from the `target` can be easily manipulated from the console. So a user can change the `svg` id in the console to be "awesome" and then the backend would try to delete "awesome".

We can remove the `contribution.id` on the `path` tags too, since id's should be unique per page.

Reply:
Sounds like a good idea will implement.